src/WPF/ArcGISRuntime.WPF.Samples/Samples/LocalServer/LocalServerGeoprocessing/LocalServerGeoprocessing.xaml.cs

```diff
@@ -0,0 +1,271 @@
+                return;
+            }
+
+            // Get the path to the geoprocessing task
+            string gpServiceUrl = await GetGpPath();
+
+            // Create the geoprocessing service
+            _gpService = new LocalGeoprocessingService(gpServiceUrl, GeoprocessingServiceType.AsynchronousSubmitWithMapServiceResult);
+
+            // Take action once the service loads
+            _gpService.StatusChanged += GpServiceOnStatusChanged;
+
+            // Try to start the service
+            try
+            {
+                // Start the service
+                await _gpService.StartAsync();
+            }
+            catch (Exception)
+            {
+                MessageBox.Show("geoprocessing service failed to start.");
+            }
+        }
+
+        private async void GpServiceOnStatusChanged(object sender, StatusChangedEventArgs statusChangedEventArgs)
+        {
+            // Return if the server hasn't started
+            if (statusChangedEventArgs.Status != LocalServerStatus.Started) return;
+
+            // Create the geoprocessing task from the service
+            _gpTask = await GeoprocessingTask.CreateAsync(new Uri(_gpService.Url + "/Contour"));
+
+            // Update UI
+            MyUpdateContourButton.IsEnabled = true;
+            MyLoadingIndicator.Visibility = Visibility.Collapsed;
+        }
+
+        private void GenerateContours()
+        {
+            // Show the progress bar
+            MyLoadingIndicator.Visibility = Visibility.Visible;
+            MyLoadingIndicator.IsIndeterminate = false;
+
+            // Create the geoprocessing parameters
+            GeoprocessingParameters gpParams = new GeoprocessingParameters(GeoprocessingExecutionType.AsynchronousSubmit);
+
+            // Add the interval parameter to the geoprocessing parameters
+            gpParams.Inputs["ContourInterval"] = new GeoprocessingDouble(MyContourSlider.Value);
+
+            // Create the job
+            _gpJob = _gpTask.CreateJob(gpParams);
+
+            // Update the UI when job progress changes
+            _gpJob.ProgressChanged += (sender, args) =>
+             {
+                 Dispatcher.Invoke(() => { MyLoadingIndicator.Value = _gpJob.Progress; });
+             };
+
+            // Be notified when the task completes (or other change happens)
+            _gpJob.JobChanged += GpJobOnJobChanged;
+
+            // Start the job
+            _gpJob.Start();
+        }
+
+        private async void GpJobOnJobChanged(object o, EventArgs eventArgs)
+        {
+            // Show message if job failed
+            if (_gpJob.Status == JobStatus.Failed)
+            {
+                MessageBox.Show("Job Failed");
+                return;
+            };
+
+            // Return if not succeeded
+            if (_gpJob.Status != JobStatus.Succeeded) { return; }
+
+            // Get the URL to the map service with the geoprocessing results
```

Comment:
Perhaps you could break line 163 into two parts and describe what is happening to create the gpServiceResultUrl string. Perhaps you might even
provide a before and after example in the comments of what the gpServiceResultUril would look like.